File: SoloLearn Codes/Convert US date to EU date/ConvertUsDateToEuDate.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Locale;
import java.util.Scanner;

public class ConvertUsDateToEuDate
{
    public static void main(String[] args) {
        String dateStr = new Scanner(System.in).nextLine();
        LocalDate date = LocalDate.parse(dateStr, DateTimeFormatter
                .ofPattern(Character.isDigit(dateStr.charAt(0)) ? 
                        "M/d/u" : "MMMM d, u", Locale.ENGLISH));
        System.out.println(date.format(DateTimeFormatter.ofPattern("d/M/yyyy")));
    }
}

File: SoloLearn Codes/Convert US date to EU date/convert_us_date_to_eu_date.swift
import Foundation

var text = readLine()!
var dateFormat = DateFormatter()
dateFormat.dateFormat = Int(String(text[text.startIndex])) != nil ? "M/d/yyyy" : "MMMM d, yyyy"
var date = dateFormat.date(from: text)!
dateFormat.dateFormat = "d/M/yyyy"
print(dateFormat.string(from: date))